lib/ContainerManager.js

```diff
@@ -23,6 +25,17 @@ Promise.longStackTraces();
 
 const SETUP_CONTEXT = "env"
 
+var log_container_events = function (container){
+  var events = ["running tasks", "tasks complete", "creating", "stopping", "restarting",
```

Comment:
Yeah, that's what I'll do - emit a specific event as well as a `stateChange` event that also includes the specific event (state) name as the first argument to the handler.